lib/assessment.rb

```diff
@@ -4,36 +4,49 @@
 # Question 1
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
+class Dog
+  def initialize(name,breed,age)
+    @name, @breed, @age = name, breed, age
+  end
 
-##
-# your code here
-##
+  attr_reader :name
+  attr_reader :breed
+  attr_reader :age
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Answer.dog = Dog.new("fido", "lab", 3)
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
+class Person
+  def initialize(name,age,location)
+    #accessor allows for read and write, is there a write only? What is it? attr_writer
+    @name, @age, @location = name, age, location
+  end
 
-##
-# your code here
-##
+  attr_writer :name
+  attr_writer :location
```

Comment:
It kills me to tell you this, Chris, but `attr_accessor` is really the one to use. Why? Because why would we ever want to be able to write to a variable but _not_ read from it? It's such a bizarre use case that `attr_writer` is almost never used.